src/hooks/useSmartAccountDerive.ts

```diff
@@ -0,0 +1,165 @@
+import {
```

Comment:
looks good, I think, most of this was over my head wrt the actual smart contract stuff. For the future but I wonder if we could pass in some params like which chainId it's for and make it compatible with any tenant rather than just derive.